simulation/simple_sensor_simulator/src/sensor_simulation/imu/imu_sensor.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tf2/LinearMath/Matrix3x3.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/LinearMath/Transform.h>
+
+#include <geometry/quaternion/euler_to_quaternion.hpp>
+#include <geometry/quaternion/quaternion_to_euler.hpp>
+#include <simple_sensor_simulator/sensor_simulation/imu/imu_sensor.hpp>
+
+namespace simple_sensor_simulator
+{
+template <>
+auto ImuSensor<sensor_msgs::msg::Imu>::generateMessage(
+  const rclcpp::Time & current_ros_time,
+  const traffic_simulator_msgs::msg::EntityStatus & status) const -> const sensor_msgs::msg::Imu
+{
+  const auto applyNoise = [&](geometry_msgs::msg::Vector3 & v) {
+    v.x += noise_distribution_(random_generator_);
+    v.y += noise_distribution_(random_generator_);
+    v.z += noise_distribution_(random_generator_);
+  };
+
+  auto imu_msg = sensor_msgs::msg::Imu();
+  imu_msg.header.stamp = current_ros_time;
+  imu_msg.header.frame_id = "tamagawa/imu_link";
+
+  auto orientation_rpy = math::geometry::convertQuaternionToEulerAngle(status.pose.orientation);
+  auto twist = status.action_status.twist;
+  auto accel = status.action_status.accel;
+
+  // Apply noise
+  if (add_noise_) {
+    applyNoise(orientation_rpy);
+    applyNoise(twist.angular);
+    applyNoise(accel.linear);
```

Comment:
It would be unnatural to add the same noise characteristics to orientation, twist, and acceleration. Could you give each of them their own noise parameters?